### PR TITLE
Remove deprecated StringUtils from WSSE custom auth provider

### PR DESCRIPTION
**Pull request by pimpreneil**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | >=2.8 |
| Fixed tickets | N/A |

### CODE REVIEW

**Comment by pimpreneil** [comment]
StringUtils class is deprecated since version 2.8 and will be removed in 3.0. Use hash_equals() instead.

File: components/security/secure_tools.rst [hunk 1]
```diff
@@ -5,22 +5,6 @@ The Symfony Security component comes with a collection of nice utilities
 related to security. These utilities are used by Symfony, but you should
 also use them if you want to solve the problem they address.
 
-Comparing Strings
-~~~~~~~~~~~~~~~~~
-
-The time it takes to compare two strings depends on their differences. This
-can be used by an attacker when the two strings represent a password for
-instance; it is known as a `Timing attack`_.
```
Comment (xabbuh) [inline components/security/secure_tools.rst]: You can now also remove this label at the end of the file.

**Comment by xabbuh** [comment]
There also is a block after the code block in the custom authentication provider chapter about the `StringUtils::equals()` method which should now be removed too.

**Comment by pimpreneil** [comment]
Thx @xabbuh . I have added the changes.

File: components/security/secure_tools.rst [hunk 2]
```diff
@@ -1,26 +1,10 @@
-Securely Comparing Strings and Generating Random Numbers
+Securely Generating Random Numbers
 ========================================================
```
Comment (xabbuh) [inline components/security/secure_tools.rst]: Can you please also shorten the underline to match the headline's length?

Comment (pimpreneil) [inline components/security/secure_tools.rst]: whoops sorry

**Comment by pimpreneil** [comment]
@xabbuh Done and squashed

**Comment by xabbuh** [comment]
:+1:

**Comment by WouterJ** [comment]
I'm not sure if we should merge this into 2.8, as `hash_equals` was introduced in PHP 5.6, while 2.8 still supports 5.3.9+

**Comment by xabbuh** [comment]
@WouterJ When using Symfony 2.8, you [get the 5.6 polyfill package automatically](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Security/Core/composer.json#L21) (Symfony already [uses it internally](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Security/Core/Util/StringUtils.php#L56)).

**Comment by WouterJ** [comment]
Tank you @pimpreneil for making sure this deprecated feature is removed from the docs completely!

I've removed the target in 3b2f4d0 and created a new PR to add a note about the polyfill, to avoid confusion: https://github.com/symfony/symfony-docs/pull/6044 If you have any comments, please tell them!

**Pull request merged**